cluster/src/main/scala/org/apache/spark/executor/SnappyExecutor.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.executor
+
+import java.net.URL
+
+import org.apache.spark.SparkEnv
+import org.apache.spark.util.{DynamicURLClassLoader, MutableURLClassLoader}
+
+class SnappyExecutor(
+    executorId: String,
+    executorHostname: String,
+    env: SparkEnv,
+    userClassPath: Seq[URL] = Nil,
+    isLocal: Boolean = false)
+    extends Executor(executorId, executorHostname, env, userClassPath , isLocal) {
+
+  private val overwriteFiles = env.conf.getBoolean("spark.files.overwrite", false)
+
+  override def createClassLoader(urls: Array[URL],
+      parentLoader: ClassLoader, userClassPathFirst: Boolean): MutableURLClassLoader = {
+    new DynamicURLClassLoader(urls, parentLoader, parentFirst = !userClassPathFirst , overwriteFiles)
+  }
+
+}
+
+
```

Comment:
I will remove the following line 
private val overwriteFiles = env.conf.getBoolean("spark.files.overwrite", false)

as it is not needed after the changes in the job server. The Job server always create a new jar file where we have added our own suffix in the name so now executor knows that the jar belongs to the same job and it can  reload it. executors can also copy this jar as it is a new jar for them.